lib/erl_type.ml

```diff
@@ -26,13 +26,16 @@ let qualifier_of_etf etf =
   qual_of_atom atom
 
 type number =
-  | IntRange of int option * int option (* (min or -infinity, max or infinity) *)
-  | IntSet of int list
+  | IntRange of {min: int_or_neg_inf; max: int_or_pos_inf}
+  | IntSet of {set: int list}
   | Integer (* integer() *)
   | Float   (* float()   *)
   | All     (* number()  *)
+and int_or_neg_inf = IntOrNegInfInt of int | IntOrNegInfNegInf
+and int_or_pos_inf = IntOrPosInfInt of int | IntOrPosInfPosInf
```

Comment:
How about

```ocaml
and min = Min of int | NegativeInfinity
and max = Max of int | Infinity
```

Reply:
Your naming looks much better than mine. I fixed to use these naming.